components/suggestion/command_provider.jsx

```diff
@@ -34,9 +40,48 @@ class CommandSuggestion extends Suggestion {
 }
 
 export default class CommandProvider {
-    handlePretextChanged(suggestionId, pretext) {
+    handlePretextChanged(pretext, resultCallback) {
         if (pretext.startsWith('/')) {
-            getSuggestedCommands(pretext.toLowerCase(), suggestionId, CommandSuggestion, pretext.toLowerCase());
+            const command = pretext.toLowerCase();
+            Client4.getCommandsList(getCurrentTeamId(store.getState())).then(
```

Comment:
This should use `getCommands` from `mattermost-redux/actions/integrations` instead of making the client call directly.

Reply:
Also, to prevent importing the global instance of the store into these, perhaps we could have the `Textbox` component bind this action itself and then pass it into the `CommandProvider` when it creates it? Alternatively, instead of having the `Textbox` in charge of binding and passing those actions, we could try turning the providers into dummy React components that are passed into the `SuggestionBox` like
```
<SuggestionBox providers={[
    <CommandProvider/>
    <EmojiProvider/>
]}/>
```
and then they could bind to the store themselves. This may be outside the scope of this ticket though, so I could create a separate ticket to look at that

Edit: Thinking about it a bit more, that second one might not work since `connect` would make the methods hard to use, and that's on top of already kind of abusing React there

Reply:
Fixing this is outside the scope of this ticket. I agree it isn't optimal, but without attaching each provider to redux, there is no way to do it because the call is async though redux.